azure-iot-device/azure/iot/device/provisioning/pipeline/pipeline_ops_provisioning.py

```diff
@@ -0,0 +1,154 @@
+
+
+class SetSymmetricKeySecurityClientArgs(PipelineOperation):
+    """
+    A PipelineOperation object which contains connection arguments which were retrieved from a
+    symmetric key security client likely by a pipeline stage which handles the
+    SetSymmetricKeySecurityClient operation.
+
+    This operation is in the group of IoTHub operations because the arguments which it accepts are
+    very specific to DPS connections and would not apply to other types of client connections
+    (such as a IotHub client).
+    """
+
+    def __init__(self, provisioning_host, registration_id, id_scope, callback=None):
+        """
+        Initializer for SetSymmetricKeySecurityClientArgs.
+        :param registration_id: The registration ID is used to uniquely identify a device in the Device Provisioning Service.
+        The registration ID is alphanumeric, lowercase string and may contain hyphens.
+        :param symmetric_key: The key which will be used to create the shared access signature token to authenticate
+        the device with the Device Provisioning Service. By default, the Device Provisioning Service creates
+        new symmetric keys with a default length of 32 bytes when new enrollments are saved with the Auto-generate keys
+        option enabled. Users can provide their own symmetric keys for enrollments by disabling this option within
+        16 bytes and 64 bytes and in valid Base64 format.
+        :param id_scope: The ID scope is used to uniquely identify the specific provisioning service the device will
+        register through. The ID scope is assigned to a Device Provisioning Service when it is created by the user and
+        is generated by the service and is immutable, guaranteeing uniqueness.
+        """
+        super(SetSymmetricKeySecurityClientArgs, self).__init__(callback=callback)
+        self.provisioning_host = provisioning_host
+        self.registration_id = registration_id
+        self.id_scope = id_scope
+        # self.symmetric_key = symmetric_key
+
+
+class SendRegistrationRequest(PipelineOperation):
+    """
+    A PipelineOperation object which contains arguments used to send a registration request
+    to an Device Provisioning Hub.
+
+    This operation is in the group of DPS operations because it is very specific to the DPS client.
+    """
+
+    def __init__(self, rid, request, callback=None):
+        """
+        Initializer for SendRegistrationRequest objects.
+
+        :param rid : The id of the request being sent
+        :param request: The request that we are sending to the service
+        :param Function callback: The function that gets called when this operation is complete or has failed.
+         The callback function must accept A PipelineOperation object which indicates the specific operation which
+         has completed or failed.
+        """
+        super(SendRegistrationRequest, self).__init__(callback=callback)
+        self.rid = rid
+        self.request = request
```

Comment:
As mentioned in another comment, these fields are kind of confusing due to the lack of encapsulation. `request` actually appears to be some kind of payload?

Reply:
yes. changed to request_body

---
In reply to: [287547711](https://github.com/Azure/azure-iot-sdk-python-preview/pull/97#discussion_r287547711) [](ancestors = 287547711)